tests/dist.rs

```diff
@@ -190,9 +190,23 @@ pub fn create_mock_channel(
         ],
     });
 
-    // An extra package that can be used as a component of the other packages
+    // Extra packages that can be used as a component of the other packages
     // for various tests
-    packages.push(bonus_component("bonus", contents));
+    packages.push(bonus_component(
+        "bonus",
+        contents.clone(),
+        "x86_64-apple-darwin",
+    ));
+    packages.push(bonus_component(
+        "bonus-2",
+        contents.clone(),
+        "x86_64-apple-darwin",
+    ));
+    packages.push(bonus_component(
+        "bonus-3",
+        contents,
+        "x86_64-unknown-linux-gnu",
+    ));
```

Comment:
Is there not an existing scenario which does what you need without adding additional components?